cargo-shuttle/src/lib.rs

```diff
@@ -1787,6 +1820,182 @@ impl Shuttle {
         Ok(CommandOutcome::Ok)
     }
 
+    async fn local_run_beta(&self, mut run_args: RunArgs) -> Result<CommandOutcome> {
+        let project_name = self.ctx.project_name().to_owned();
+        let services = self.pre_local_run(&run_args).await?;
+        let service = services
+            .first()
+            .expect("at least one shuttle service")
+            .to_owned();
+
+        trace!(path = ?service.executable_path, "runtime executable");
+
+        let secrets = Shuttle::get_secrets(&run_args, &service)?;
+        Shuttle::find_available_port_beta(&mut run_args);
+        Shuttle::check_and_warn_runtime_version(&service.executable_path).await?;
+
+        let runtime_executable = service.executable_path.clone();
+        let api_port = portpicker::pick_unused_port()
+            .expect("failed to find available port for local provisioner server");
+        let api_addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), api_port);
+        let ip = if run_args.external {
+            Ipv4Addr::UNSPECIFIED
+        } else {
+            Ipv4Addr::LOCALHOST
+        };
+
+        let state = Arc::new(ProvApiState {
+            project_name,
+            secrets,
+        });
+        let make_svc = make_service_fn(move |_conn| {
+            let state = state.clone();
+            async {
+                Ok::<_, Infallible>(service_fn(move |req| {
+                    let state = state.clone();
+                    handler(state, req)
+                }))
+            }
+        });
+        let server = Server::bind(&api_addr).serve(make_svc);
+        tokio::spawn(async move {
+            if let Err(e) = server.await {
+                eprintln!("Provisioner server error: {}", e);
+                exit(1);
+            }
+        });
+
+        println!(
+            "\n    {} {} on http://{}:{}\n",
+            "Starting".bold().green(),
+            service.package_name,
+            ip,
+            run_args.port,
+        );
+
+        info!(
+            path = %runtime_executable.display(),
+            "Spawning runtime process",
+        );
+        let mut runtime = tokio::process::Command::new(
+            dunce::canonicalize(runtime_executable).context("canonicalize path of executable")?,
+        )
+        .current_dir(&service.workspace_path)
+        .args(["--run"])
+        .envs([
+            ("SHUTTLE_BETA", "true"),
+            ("SHUTTLE_PROJECT_ID", "proj_LOCAL"),
+            ("SHUTTLE_PROJECT_NAME", "TODO"),
```

Comment:
What is the plan for this, and does the user see it?

Reply:
TY for reminder. It's passed into ResourceFactory and Metadata.